### PR TITLE
Fix xdebug handling

### PR DESCRIPTION
**Pull request by staabm**
before this PR 

- the debugger halts in the main process even without the `--xdebug` flag (opposite of how PHPStan behaves)
- the debugger does **not** halt in the main process when `--xdebug` flag is passed (opposite of how PHPStan behaves)

after this PR it behaves like PHPStan, see https://github.com/phpstan/phpstan-src/blob/29a68b75ce9fa885886395df1acd23c06cff2a14/src/Command/CommandHelper.php#L99-L104

refs https://github.com/rectorphp/rector/issues/8060#issuecomment-1920823632

---

so to sum up: if you want to debug something with the step-debugger you need to invoke rector with `--xdebug` option. thats expected and thats how PHPStan behaves.

just having `xdebug` extension loaded is not enough.

### CODE REVIEW

File: src/Console/ConsoleApplication.php [hunk 1]
```diff
@@ -41,10 +41,10 @@ public function __construct(array $commands)
 
     public function doRun(InputInterface $input, OutputInterface $output): int
     {
-        // @fixes https://github.com/rectorphp/rector/issues/2205
```
Comment (staabm) [inline src/Console/ConsoleApplication.php:44]: removed this comment as I don't see how it makes sense with the current code

File: src/Console/ConsoleApplication.php [hunk 2]
```diff
@@ -41,10 +41,10 @@ public function __construct(array $commands)
...
         $isXdebugAllowed = $input->hasParameterOption('--xdebug');
-        if ($isXdebugAllowed) {
+        if (!$isXdebugAllowed) {
```
Comment (samsonasik) [inline src/Console/ConsoleApplication.php:45]: I am curious why negation is needed?

Comment (staabm) [reply src/Console/ConsoleApplication.php:45]: Calling `xDebugHandler->check()` will restart the process when xdebug is loaded. Therefore we should only call into this method when xdebug is not allowed

Comment (samsonasik) [reply src/Console/ConsoleApplication.php:45]: what I mean is, `hasParameterOption()` verify that `--xdebug` exists in the passed parameter, eg: `vendor/bin/rector --xdebug`.

when using negation `! $isXdebugAllowed` checked, it means that `--xdebug` is never passed in cli parameter.

If it was working on non-parallel, and not working in parallel, I guess the issue is on visibility of `--xdebug` in parallel env.

**Comment by samsonasik** [comment]
while i don't really understand why, if it works, it works, let's give it a try, thank you @staabm

File: src/Console/ConsoleApplication.php (hunk 2 above)
Comment (staabm) [reply src/Console/ConsoleApplication.php:45]: Before this PR the worker command already contained the `--xdebug` flag. 

The code change here is related to how rector behaves when `--xdebug` is passed, which was inverted before this PR
